tiny_dnn/layers/deconvolutional_layer.h

```diff
@@ -158,66 +156,21 @@ class deconvolutional_layer : public feedforward_layer<Activation> {
                         serial_size_t w_stride = 1,
                         serial_size_t h_stride = 1,
                         backend_t backend_type = core::default_engine())
-    : Base(std_input_order(has_bias)) {
+    : layer(std_input_order(has_bias), {vector_type::data}) {
     deconv_set_params(shape3d(in_width, in_height, in_channels), window_size,
                       window_size, out_channels, pad_type, has_bias, w_stride,
                       h_stride, connection_table);
     init_backend(backend_type);
   }
 
-  /**
-  * constructing deconvolutional layer
-  *
-  * @param in_width         [in] input image width
```

Comment:
@edgarriba: The error stated that this constructor definition doesn't match with anyone from `layer`. I thought it was derived from `feedforward_layer`